### PR TITLE
[sp-solutions][HTTPRequestCountFilter]Upgrade search bar in request count perspective

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
The autocomplete used in request count perspective search bar is upgraded to material ui 1.5. The drop down is rendered as a pop over so that it doesn't get hidden behind widgets beneath it.

### CODE REVIEW

**Comment by TanyaM** [comment]
Please refactor as per code review changes.

**Comment by ruwiniwj** [comment]
Fixed as per the code review suggestions

File: components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/src/HTTPAnalyticsRequestCountFilter.jsx [hunk 1]
```diff
@@ -270,17 +425,45 @@ class HTTPAnalyticsRequestCountFilter extends Widget {
...
+            selectedSingleServiceValue: userSelection.method,
         }, this.publishUpdate);
     }
 
+    /**
+     * Sort in ascending order
+     * @param a
+     * @param b
+     */
+    naturalSort(a, b) {
```
Comment (TanyaM) [inline components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/src/HTTPAnalyticsRequestCountFilter.jsx]: Change as suggested in code review

File: components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/package.json [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
+        "@material-ui/icons": "2.0.3",
+        "@wso2-dashboards/widget": "1.4.0",
         "prop-types": "15.6.1",
-        "react": "16.3.2",
+        "react": "16.4.0",
         "react-custom-scrollbars": "4.2.1",
-        "react-dom": "16.3.2",
-        "react-select": "2.0.0-beta.4",
+        "react-dom": "16.4.0",
+        "react-select": "2.1.0",
```
Comment (TanyaM) [inline components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/package.json]: Is the feature tested from 2.0.0-beta.4 move to 2.1.0?

File: components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/src/HTTPAnalyticsRequestCountFilter.jsx (hunk 1 above)
Comment (ruwiniwj) [reply components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/src/HTTPAnalyticsRequestCountFilter.jsx]: This is the comparator function passed to the sort method, not a sort function.

File: components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/package.json (hunk 2 above)
Comment (ruwiniwj) [reply components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/package.json]: Yes.

**Pull request merged**